src/features/transfer-ownership/useTransferOwnershipForm.test.tsx

```diff
@@ -0,0 +1,255 @@
+import React from 'react';
+
+import { render, waitFor } from '@testing-library/react';
+
+import {
+	useTransferOwnershipForm,
+	UseTransferOwnershipFormReturn,
+} from './useTransferOwnershipForm';
+import { Step } from './TransferOwnership.constants';
+import * as mock from './TransferOwnership.mocks';
+
+// mock transferDomainOwnership()
+var mockTranferOwnership = jest.fn();
+
+// mock transferDomainOwnership() transaction
+var mockTx = jest.fn();
+
+// mock sdk
+jest.mock('../../lib/hooks/useZnsSdk', () => ({
+	useZnsSdk: () => ({
+		transferDomainOwnership: mockTranferOwnership,
+	}),
+}));
+
+// mock web3 - provider & account
+jest.mock('../../lib/hooks/useWeb3', () => ({
+	useWeb3: () => ({
+		provider: {
+			getSigner: () => ({ isSigner: true }),
+		},
+		account: mock.connectedAccount,
+	}),
+}));
+
+// setup testing
+const jestConsole = console;
+const consoleError = console.error;
+
+beforeEach(() => {
+	global.console = require('console');
+	console.error = jest.fn();
+});
+
+afterEach(() => {
+	global.console = jestConsole;
+	console.error = consoleError;
+	jest.clearAllMocks();
+});
+
+// setup hook
+const setupHook = (domainId: string, domainOwner: string) => {
+	const returnVal = {};
+	const TestComponent = () => {
+		Object.assign(returnVal, useTransferOwnershipForm(domainId, domainOwner));
+		return null;
+	};
+	render(<TestComponent />);
+	return returnVal as UseTransferOwnershipFormReturn;
+};
+
+describe('useTransferOwnershipForm', () => {
```

Comment:
These tests look good - there are some changes that I think _could_ make them more robust but I like what I see 🔥

At the moment we're testing `useTransferOwnershipForm` as an isolated unit. For features like this, I like to [test from the user's perspective](https://kentcdodds.com/blog/how-to-know-what-to-test), as one of the main goals of testing is to gain confidence that a feature works for users.

I wrote [the `CancelBid` tests](https://github.com/zer0-os/zNS-dapp/blob/develop/src/containers/flows/CancelBid/CancelBid.test.tsx) in this style. These tests mock the required SDK functions, then use the form like a user would. Because of this, it doesn't matter how `CancelBid` works under the hood, it just matters that the user can navigate through the flow and execute the transaction successfully.

_Note: I can also see ways to improve the `CancelBid` tests, they're not perfect but provide a decent guide. I also think the `useCancelBid.test.tsx` tests were added in the zAuction 2.1 update but I don't think they need to be there - they're double ups of the `CancelBid` tests._

A bit of faff above, but I think these tests would be made more robust if you:
1. Move these tests to `TransferOwnershipForm.test.tsx`.
2. Adjust the tests to be validated by the `Transfer Ownership` form UI,  rather than asserting state/return values.

I'm not sure if the above really is the best approach, but it's how I like to write tests and I think it makes the most sense. Also, if the above doesn't make sense, let me know and I'll write the tests and we can review them together!

Reply:
Awesome thanks for this write up @colbr, helps my understanding a lot. I completely get what you're saying above, when I was writing the tests and thinking of the follow up Form tests that I would be writing I was foreseeing a lot of duplication. I agree that if we assert the flow of the form (happy and unhappy paths) then that will cover ```useTransferOwnershipForm``` and the actual navigation for the user. 

I'll have a crack at updating these today, fighting some config issues at the moment but making progress. I think the package-lock.json stuff might be in a bit of trouble, will note down any issues I spot that I'm unsure of.